aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -172,28 +201,45 @@ SparseTensor& asin_sparse_(SparseTensor& t) {
 
 // TODO: add in-place variant
 
-SparseTensor& sqrt_out_sparse(SparseTensor& r, const SparseTensor& t_) {
-  TORCH_CHECK(r.is_sparse(), "Tensor should be sparse");
-  TORCH_CHECK(t_.is_sparse(), "Tensor should be sparse");
-
-  // This coalesce is why we can't easily provide an inplace variant
-  SparseTensor t = t_.coalesce();
-
-  r.resize_as_(t);
-  auto indices = r._indices();
-  indices.resize_as_(t._indices());
-  indices.copy_(t._indices());
-  Tensor r_values = r._values();
-  at::sqrt_out(r_values, t._values());
-  get_sparse_impl(r)->set_nnz_and_narrow(t._nnz());
-  return r._coalesced_(t.is_coalesced());
+SparseTensor& sqrt_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  sparse_unary_op_check_sanity(r, t, true);
```

Comment:
change `true` -> `false` as sqrt is not linear.

Reply:
This true is for promotes_integer_to_float; not for is_linear_map; have updated with a comment for greater clarity

Reply:
thanks!